### ISSUE TITLE
PYI015 covers parts of PYI052

### ISSUE THREAD

**Issue by twoertwein** [charliermarsh/ruff#4227]
flake8-pyi
> Y015 | Only simple default values are allowed for assignments. Similar to Y011, but for assignments rather than parameter annotations.
> -- | --
> Y052 | Y052 disallows assignments to constant values where the assignment does not have a type annotation. For example, x = 0 in the global namespace is ambiguous in a stub, as there are four different types that could be inferred for the variable x: int, Final[int], Literal[0], or Final[Literal[0]]. Enum members are excluded from this check, as are various special assignments such as __all__ and __match_args__.

ruff seems to include Y52 in Y15 but does not exclude enums.

in pyi file:
```py
from enum import Enum

class _NoDefault(Enum):
    no_default = ...

no_default: Final = _NoDefault.no_default  # noqa: PYI015
NoDefault = Literal[_NoDefault.no_default]
```

xref https://github.com/pandas-dev/pandas/pull/53062#discussion_r1185161948

**Comment by charliermarsh** [comment]
👍 I think the best solution here is to implement `Y052`, and fix `Y015` such that they're non-overlapping (as in `flake8-pyi`).

**Comment by sladyn98** [comment]
@charliermarsh I would like to work on this issue, can you point me to some code pointers and I can work on it.

**Comment by mccullocht** [comment]
@sladyn98 I briefly looked into this. The [pylint](https://github.com/PyCQA/flake8-pyi/blob/7425435c81d852a3c5a4be45a5b714785cec4ce1/pyi.py#L996) implementation of `Y052` and `Y015` appear together, it probably makes sense structure things similarly and place it near the [ruff](https://github.com/charliermarsh/ruff/blob/539af34f585eee4d762f6bd01ca508429c35d85e/crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs#L396) implementation of `Y015`.

**Comment by charliermarsh** [comment]
Closed by https://github.com/charliermarsh/ruff/pull/4293.
